packages/legend-studio/src/stores/editor-state/GraphGenerationState.ts

```diff
@@ -128,14 +127,22 @@ export class GraphGenerationState {
   get supportedFileGenerationConfigurationsForCurrentElement(): GenerationConfigurationDescription[] {
     if (this.editorStore.currentEditorState instanceof ElementEditorState) {
       const currentElement = this.editorStore.currentEditorState.element;
-      if (
-        currentElement instanceof Class ||
-        currentElement instanceof Enumeration
-      ) {
-        return this.fileGenerationConfigurations
-          .slice()
-          .sort((a, b): number => a.label.localeCompare(b.label));
-      }
+      const getExtraFileGenerationScopeFilters = this.editorStore.pluginManager
```

Comment:
:) noble idea, but I think this will cause old use case to fail, we still need somebody to say `Class` and `Enumeration` are fine right?

Reply:
Thanks! This should be fine for `Class` and `Enumeration` since the filter checks for each generation type e.g. Morphir, and only filters out entity types e.g. `Class` if matching, so `Class` and `Enumeration` will not be removed for other generation types. I think technically if we want to be more methodical we could add additional filters for other generation types.

Reply:
That's fair, but I don't think it's wise to break the old behavior without much consideration here. I believe this would mean we now allow `Avro`, `JSON Schema` for things like `Mapping`. Now that's not great. I think what we should do is to add the old scope filter back here. We can easily add one more item to `getExtraFileGenerationScopeFilters` right?